src/components/Swap.vue

```diff
@@ -259,103 +246,92 @@ export default class Swap extends Mixins(TranslationMixin, LoadingMixin) {
-            throw error
-          }
-        }
-      }
-      this.setToValue(this.formModel.to)
+    if (value !== this.fromValue) {
+      this.setFromValue(value)
+      await this.recountSwapValues()
     }
-    this.setFromValue(this.formModel.from)
   }
 
   async handleInputFieldTo (value): Promise<any> {
-    this.formModel.to = value
+    if (!this.areTokensSelected || this.isZeroValue(value)) {
+      this.resetFieldFrom()
+    }
 
-    if (!this.isFieldFromActive) {
-      this.isFieldToActive = true
-      if (!this.areTokensSelected || this.isZeroToAmount) {
-        this.resetFieldFrom()
-      } else {
-        try {
-          // Always use getSwapResult and minMaxReceived with reversed flag for Token B
-          const isExchangeBSwap = true
-          const swapResult = await api.getSwapResult(this.tokenFrom.address, this.tokenTo.address, this.formModel.to, isExchangeBSwap)
-          this.formModel.from = swapResult.amount
-          this.setLiquidityProviderFee(swapResult.fee)
-          const minMaxReceived = await api.getMinMaxValue(this.tokenFrom.address, this.tokenTo.address, swapResult.amount, this.slippageTolerance, isExchangeBSwap)
-          this.setMinMaxReceived({ minMaxReceived, isExchangeB: isExchangeBSwap })
-          this.updatePrices()
-          this.resetInsufficientAmountFlag()
-          if (this.connected) {
-            await this.getNetworkFee()
-            if (this.tokenFrom.symbol !== KnownSymbols.XOR) {
-              await this.getTokenXOR()
-            }
-          }
-        } catch (error) {
-          this.resetFieldFrom()
-          if (!this.isInsufficientAmountError(this.tokenTo.symbol, error.message)) {
-            throw error
-          }
-        }
-      }
-      this.setFromValue(this.formModel.from)
+    if (value !== this.toValue) {
+      this.setToValue(value)
+      await this.recountSwapValues()
     }
-    this.setToValue(this.formModel.to)
   }
 
-  async recountSwapValues (): Promise<void> {
-    if (this.isFieldFromActive) {
-      await this.handleInputFieldFrom(this.fromValue)
-    } else {
-      await this.handleInputFieldTo(this.toValue)
+  private async recountSwapValues (): Promise<void> {
+    const value = this.isExchangeB ? this.toValue : this.fromValue
+    const setOppositeValue = this.isExchangeB ? this.setFromValue : this.setToValue
+    const resetOppositeValue = this.isExchangeB ? this.resetFieldFrom : this.resetFieldTo
+    const token = this.isExchangeB ? this.tokenTo : this.tokenFrom
+
+    if (!this.areTokensSelected || this.isZeroValue(value)) return
+
+    try {
+      this.isRecountingProcess = true
+      this.resetInsufficientAmountFlag()
+
+      const { amount, fee } = await api.getSwapResult(this.tokenFrom.address, this.tokenTo.address, value, this.isExchangeB)
+
+      setOppositeValue(amount)
+      this.setLiquidityProviderFee(fee)
+
+      await Promise.all([
```

Comment:
is it ok here if any of promise will be rejected (cuz another promises won't be finished)?

Reply:
Yes, all of these calculations are independent of each other, and rejecting any of them does not affect the others